### PR TITLE
Fix perftests CMake target being empty: https://github.com/GridTools/gridtools/issues/1551

### LINKED ISSUE #1551: CMake: errors out on perftest if neither OpenMP nor CUDA was found
because the perftest target doesn't contain SOURCES.

### PR DESCRIPTION
**Pull request by petiaccja**
Fixes #1551. You cannot make a CMake target with no sources and no linker input, so this adds a dummy source file to the perftests target for when there are no linker inputs either.

### CODE REVIEW

File: tests/regression/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,11 @@
 add_executable(perftests)
 
+file(WRITE 
+     "${CMAKE_BINARY_DIR}/perftests_dummy.cpp"
```
Comment (havogt) [inline tests/regression/CMakeLists.txt:4]:
```suggestion
     "${CMAKE_CURRENT_BINARY_DIR}/perftests_dummy.cpp"
```

File: tests/regression/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
+     "// This file is needed in some cases because CMake targets must have at least one .cpp source."
+)
+target_sources(perftests PRIVATE "${CMAKE_BINARY_DIR}/perftests_dummy.cpp")
```
Comment (havogt) [inline tests/regression/CMakeLists.txt:7]:
```suggestion
target_sources(perftests PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/perftests_dummy.cpp")
```

**Review by havogt — CHANGES_REQUESTED** [review]
Just a minor thing: let's put it in the current build dir, i.e. in <build>/tests/regression.

**Review by havogt — APPROVED** [review]
Thanks. Looks good (as good as CMake allows).

**Comment by petiaccja** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by petiaccja** [comment]
launch jenkins

**Comment by petiaccja** [comment]
launch jenkins

**Comment by petiaccja** [comment]
launch jenkins

**Comment by petiaccja** [comment]
launch jenkins

**Comment by petiaccja** [comment]
launch perftest

**Pull request merged**